### ISSUE TITLE
Can't Build Due to Outdated Java 8 Dependency

### ISSUE THREAD

**Issue by ChthonVII** [GoMule/GoMule#4]
Some recent addition (Kotlin?) has pulled in a hard dependency on tools.jar, which was deprecated in Java 9. So GoMule simply won't build on any version of Java newer than 8. This is a showstopper for much of the Linux world, as most distributions dropped the Java 8 JDK years ago.

**Comment by ajgringo619** [comment]
Does the pre-compiled jar run? I'm using it with Java 18 on Linux.

**Comment by ChthonVII** [comment]
Yes, the jar seems to be forward compatible once it's built. Building it requires outdated Java 8 stuff.

Also, I found a workaround: You can download unpackaged outdated JRE/JDKs from Eclipse's OpenJDK fork from: [https://adoptium.net/temurin/releases](https://adoptium.net/temurin/releases). Then set JAVA_HOME to force the build process to use that instead of the system Java.

**Comment by silospen** [comment]
I'm unsure of the issues you're having here, it compiles fine for me with both Java 11 and Java 16 JDKs. You should not have to downgrade your JDK version. Can you provide any kind of output to show what's going wrong?

**Comment by ChthonVII** [comment]
```
git clone https://github.com/GoMule/GoMule.git
cd GoMule/GoMule
./gradlew build
```
gives
`
ERROR: Could not find tools.jar`
Because tools.jar doesn't exist in any version of Java later than 8.

It appears this is some sort of dependency-hell problem, since whatever wanted tools.jar is getting cached somewhere (gradle cache I presume), so the problem doesn't recur once you've got that bit built. Since this is a new problem since I forked back in November, it's got to be a dependency you've added since then.

**Comment by ouflak** [comment]
> I'm unsure of the issues you're having here, it compiles fine for me ...

Probably because you at some point installed the Java version 8 sdk and still have that environment lingering around. You might have subsequent versions installed, but as long that sdk and its environment variable are on your system, you will be able to compile. For those of us who have never had Java 8 (I just now installed 18 fresh on this machine), we're running into this dependency issue. I recommend that you refactor your code to live without that dependency, especially as it's long since been deprecated.

**Comment by silospen** [comment]
Believe me, if I knew what the dependency was I'd look into it. At the moment I still can't seem to reproduce:
 - I only have Coretto 11 and openJDK 16 on this machine, it builds fine with either
 - I've never installed Java 8 on this machine
 - I searched my hard drive and there's no sign of a tools.jar hanging out anywhere
 - Deleted my gradle folder (so cache) and rebuilt from scratch

And it's all just working fine. I guess next I'll have to get a clean linux install and start from there? Any other pointers from either of you?

**Comment by ChthonVII** [comment]
Had I known the issue would become unreproducible after compiling once with Java 8, I would have copy/pasted the full output of the build command. Unfortunately, I was too focused on getting it to work to anticipate that possibility.

The only help I'm able to provide without doing a clean linux install myself is that this issue is relatively new. It didn't happen when I forked R.4.4.13 back in November.
